lib/types.js

```diff
@@ -18,14 +19,16 @@ const exclude = function (types, excludes) {
     return types;
   }
 
-  for (const doc of types) {
+  const typesCopy = cloneDeep(types);
+
+  for (const doc of typesCopy) {
     for (const def of doc.definitions) {
       if (def.kind !== 'ObjectTypeDefinition') {
         continue;
       }
       if (['Query', 'Mutation', 'Subscription'].indexOf(def.name.value) > -1) {
         def.fields = def.fields.filter((field) => {
-          if (!check(def.name.value, field.name.value, excludes)) {
+          if (check(def.name.value, field.name.value, excludes)) {
```

Comment:
removed the `!` since this function now returns `true` when a field should be excluded.